scipy/stats/_distribution_infrastructure.py

```diff
@@ -966,8 +997,11 @@ def filtered(self, x, *args, **kwargs):
                             else np.any(mask_endpoint))
 
         # Check for non-integral arguments to PMF method
+        # or PDF of a discrete distribution.
         any_non_integral = False
-        if method_name in replace_non_integral:
+        if (
+                method_name in replace_non_integral
+                or discrete and method_name in {"pdf", "logpdf"}):
```

Comment:
Non-integral arguments to `pdf` of a discrete distribution should be replaced with zero too. I guess we could also just check if the value is out of the support, and have different out-of-support replace logic for discrete vs continuous distributions. I think this was the easier fix, though I'm a little concerned it makes things harder to understand.

Reply:
I confirmed that this wasn't working correctly before. What probably happened is that I added `pdf` to discrete distributions (and `pmf` to continuous distributions) after upating `filtered`.

If I am thinking straight, PMF for continuous distributions is *always* zero - I think we discussed not considering singularities of continuous distributions. Also, your logic `or discrete and method_name in {"pdf", "logpdf"}):` is only done for discrete distributions. So we only need to make this replacement for `discrete` distributions.

So can we just add `pdf` and `logpdf` to the set of `replace_non_integral` and make it:
```suggestion
        if discrete and method_name in replace_non_integral:
```
?